spacetime/tests/test_linear_scalar.py

```diff
@@ -133,4 +134,90 @@ def _march():
+    or check if the solver's mathematical model is correct.
+    """
+    
+    @staticmethod
+    def _build_solver(resolution):
+        grid = libst.Grid(0, 4 * 2 * np.pi, resolution)
+        cfl = 0.99
+        dx = (grid.xmax - grid.xmin) / grid.ncelm
+        dt = dx * cfl
+        svr = libst.LinearScalarSolver(grid=grid, time_increment=dt)
+
+        # Initialize
+        for e in svr.selms(odd_plane=False):
+            if e.xctr < 2 * np.pi or e.xctr > 4 * np.pi:
+                v = 0
+                dv = 0
+            else:
+                v = np.sin(e.xctr)
+                dv = np.cos(e.xctr)
+            e.set_so0(0, v)
+            e.set_so1(0, dv)
+
+        svr.setup_march()
+
+        return svr
+
+    @staticmethod
+    def _exact_solution(svr, iter_num, gfun):
+        v = []
+        for e in svr.selms(odd_plane=False):
+            x = (
+                e.xctr - iter_num * svr.dt
+            ) % svr.grid.xmax  # dealt with boundary treatment
+            if x < 2 * np.pi or x > 4 * np.pi:
+                v.append(0)
+            else:
+                v.append(gfun(x))
+        return np.array(v)
+
+    def setUp(self):
+        self.resolution = 256
+        self.svr = self._build_solver(self.resolution)
+        self.cycle = range(1001)
+
+    def test_grid_test(self):
+        _norm = lambda vec, ord: sum(np.abs(vec) ** ord) ** (1 / ord)
+
+        grid_num = [320, 640, 1280, 2560, 5120, 10240, 20480, 
+                    40960, 81920, 163840, 327680]
+        dx = []
+        err = []
+        it_num = 20
+        for grid in grid_num:
+            svr = self._build_solver(grid)
+            svr.march_alpha2(it_num)
+            exact_sol = self._exact_solution(svr, it_num, np.sin)
+            dx.append(svr.grid.ncelm)
+            err.append(_norm(svr.get_so0(0).ndarray - exact_sol, 1))
+        
+        self.assertAlmostEqual(1.0, abs(math.log(abs(err[0] / err[1]))
+                                / math.log(abs(dx[0] / dx[1]))), places=1)
+        self.assertAlmostEqual(1.0, abs(math.log(abs(err[1] / err[2])) 
+                                / math.log(abs(dx[1] / dx[2]))), places=1)
+        self.assertAlmostEqual(1.1, abs(math.log(abs(err[2] / err[3])) 
+                                / math.log(abs(dx[2] / dx[3]))), places=1)
+        self.assertAlmostEqual(1.0, abs(math.log(abs(err[3] / err[4]))
+                                / math.log(abs(dx[3] / dx[4]))), places=1)
+        self.assertAlmostEqual(1.0, abs(math.log(abs(err[4] / err[5])) 
+                                / math.log(abs(dx[4] / dx[5]))), places=1)
+        self.assertAlmostEqual(1.0, abs(math.log(abs(err[5] / err[6])) 
+                                / math.log(abs(dx[5] / dx[6]))), places=1)
+        self.assertAlmostEqual(0.87, abs(math.log(abs(err[6] / err[7])) 
+                                / math.log(abs(dx[6] / dx[7]))), places=1)
```

Comment:
Should it be 'places=2'?

Reply:
I forgot to change this condition in previous commit... 
I fixed it in https://github.com/yungyuc/turgon/pull/32/commits/f2f9db6b3cd63da7ed2ec7766cc877d24d54775f